assets/js/ajax-filters.js

```diff
@@ -1,288 +1,458 @@
 /* global job_manager_ajax_filters */
-jQuery( document ).ready( function ( $ ) {
+jQuery( document ).ready( function( $ ) {
+	function job_manager_supports_html5_session_storage() {
+		return window.sessionStorage && typeof window.sessionStorage.setItem === 'function';
+	}
 
-	var xhr = [];
+	var session_storage_prefix = 'job_listing_';
 
-	$( '.job_listings' ).on( 'update_results', function ( event, page, append, loading_previous ) {
-		var data         = '';
-		var target       = $( this );
-		var form         = target.find( '.job_filters' );
-		var showing      = target.find( '.showing_jobs' );
-		var results      = target.find( '.job_listings' );
-		var per_page     = target.data( 'per_page' );
-		var orderby      = target.data( 'orderby' );
-		var order        = target.data( 'order' );
-		var featured     = target.data( 'featured' );
-		var filled       = target.data( 'filled' );
-		var job_types    = target.data( 'job_types' );
-		var post_status  = target.data( 'post_status' );
-		var index        = $( 'div.job_listings' ).index(this);
-		var categories, keywords, location;
-
-		if ( index < 0 ) {
-			return;
+	function job_manager_store_state( $target, state ) {
+		if ( ! job_manager_supports_html5_session_storage() ) {
+			return false;
 		}
 
-		if ( xhr[index] ) {
-			xhr[index].abort();
+		if ( typeof state !== 'object' ) {
+			state = {};
 		}
 
-		if ( true === target.data( 'show_filters' ) ) {
+		var $form = $target.find( '.job_filters' );
+		var index = $( 'div.job_listings' ).index( $target );
 
-			var filter_job_type = [];
+		state.form = $form.serialize();
 
-			$( ':input[name="filter_job_type[]"]:checked, :input[name="filter_job_type[]"][type="hidden"], :input[name="filter_job_type"]', form ).each( function () {
-				filter_job_type.push( $( this ).val() );
-			} );
+		var session_storage_key = session_storage_prefix + index;
 
-			categories = form.find( ':input[name^="search_categories"]' ).map( function () {
-				return $( this ).val();
-			} ).get();
-			keywords   = '';
-			location   = '';
-			var $keywords  = form.find( ':input[name="search_keywords"]' );
-			var $location  = form.find( ':input[name="search_location"]' );
-
-			// Workaround placeholder scripts
-			if ( $keywords.val() !== $keywords.attr( 'placeholder' ) ) {
-				keywords = $keywords.val();
-			}
+		try {
+			return window.sessionStorage.setItem( session_storage_key, JSON.stringify( state ) );
+		} catch ( e ) {}
```

Comment:
Could we add a comment saying why it's ok to squelch the exception here?

Reply:
Done in 38b06765d730da12f993c82916a9ba82c9d3eb3f